docs/src/reference/variables.md

```diff
@@ -2,6 +2,18 @@
 
 Variables are part of a workflow instance and represent the data of the instance. A variable has a name and a JSON value. The visibility of a variable is defined by its variable scope.
 
+## Variable Names
+
+The name of a variable can be any alphanumeric string including the `_` symbol. For a combination of words, it is recommended to use the `camelCase` or the `snake_case` format.
```

Comment:
Perhaps we should additionally note here that `kebab-case` cannot be used, because it might be a common pitfall.